src/main/kotlin/uk/gov/justice/digital/hmpps/incentivesapi/util/flow/toMap.kt

```diff
@@ -0,0 +1,15 @@
+package uk.gov.justice.digital.hmpps.incentivesapi.util.flow
+
+import kotlinx.coroutines.flow.Flow
+
+/**
+ * Collects given flow into a [destination] map indexed by the key returned from [keySelector] function
+ * applied to each element. If any two elements would have the same key returned by [keySelector]
+ * the last one gets added to the map.
+ */
+suspend fun <K, V> Flow<V>.toMap(destination: MutableMap<K, V> = LinkedHashMap(), keySelector: (V) -> K): Map<K, V> {
```

Comment:
coroutines library doesn't provide a `.toMap` but it’s a natural extension like
https://github.com/Kotlin/kotlinx.coroutines/blob/6dfabf763fe9fc91fbb73eb0f2d5b488f53043f1/kotlinx-coroutines-core/common/src/flow/terminal/Collection.kt#L20
using the map associating concept of the standard library
https://github.com/JetBrains/kotlin/blob/c04e1b6ce705c74b30821804824457d02d1a4538/libraries/stdlib/common/src/generated/_Collections.kt#L1193